staging/src/k8s.io/kube-aggregator/pkg/controllers/autoregister/autoregister_controller.go

```diff
@@ -169,29 +197,61 @@ func (c *autoRegisterController) processNextWorkItem() bool {
 	return true
 }
 
-func (c *autoRegisterController) checkAPIService(name string) error {
+// checkAPIService syncs the current APIService against a list of desired APIService objects
+//
+// Current:           \          Desired: | A. not found | B. sync on start | C. sync always
+// ---------------------------------------|--------------|------------------|---------------
+// 1. error                               | error        | error            | error
+// 2. not found                           | -            | create once      | create
+// 3. no sync                             | -            | -                | -
+// 4. sync on start, not present at start | -            | -                | -
```

Comment:
is "not present at start" desired or actual state?  I think its desired, but that wouldn't match the table header.

Reply:
x column was desired, y column was current. fixed the headers to clarify.